service_configuration_lib/spark_config.py

```diff
@@ -1089,12 +1089,17 @@ def get_spark_conf(
             # We want to make the app name more unique so that we can search it
             # from history server.
             app_name = f'{app_base_name}_{ui_port}_{int(time.time())}'
+        is_jupyter = _is_jupyterhub_job(app_name)
 
         # Explicitly setting app id: replace special characters to '_' to make it consistent
         # in all places for metric systems:
         # - since in the Promehteus metrics endpoint those will be converted to '_'
         # - while the 'spark-app-selector' executor pod label will keep the original app id
-        app_id = re.sub(r'[\.,-]', '_', app_name)
+        if is_jupyter:
```

Comment:
Maybe add a TODO comment here to also change for non-jupyter pods with the logic we discussed.